proto/prysm/v1alpha1/wrapper/mutator.go

```diff
@@ -0,0 +1,77 @@
+package wrapper
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	eth "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1/block"
+	"github.com/prysmaticlabs/prysm/runtime/version"
+)
+
+type BlockMutator struct {
+	Phase0    func(beaconBlock *eth.SignedBeaconBlock)
+	Altair    func(beaconBlock *eth.SignedBeaconBlockAltair)
+	Bellatrix func(beaconBlock *eth.SignedBeaconBlockBellatrix)
+}
+
+func (m BlockMutator) Apply(b block.SignedBeaconBlock) error {
+	switch b.Version() {
+	case version.Phase0:
+		bb, err := b.PbPhase0Block()
+		if err != nil {
+			return err
+		}
+		m.Phase0(bb)
+		return nil
+	case version.Altair:
+		bb, err := b.PbAltairBlock()
+		if err != nil {
+			return err
+		}
+		m.Altair(bb)
+		return nil
+	case version.Bellatrix:
+		bb, err := b.PbBellatrixBlock()
+		if err != nil {
+			return err
+		}
+		m.Bellatrix(bb)
+		return nil
+	}
+	msg := fmt.Sprintf("version %d = %s", b.Version(), version.String(b.Version()))
+	return errors.Wrap(ErrUnsupportedSignedBeaconBlock, msg)
+}
+
+func SetBlockStateRoot(b block.SignedBeaconBlock, sr [32]byte) error {
```

Comment:
Im liking the direction of this pattern a lot. Perhaps we can turn this into a tracking issue for us to use this for all the getters of the different beacon blocks? I can help with a tracking issue